src/cirq_iqm/iqm_device.py

```diff
@@ -214,63 +183,105 @@ def decompose_operation(self, operation: cirq.Operation) -> cirq.OP_TREE:
             on_stuck_raise=None
         )
 
-    @staticmethod
-    def simplify_circuit(circuit: cirq.Circuit) -> None:
-        """Simplifies and optimizes the given circuit (in place).
+    def route_circuit(self, circuit: cirq.Circuit) -> cirq.Circuit:
+        """Routes the given circuit to the device connectivity.
+
+        The routed circuit uses the device qubits, and may have additional SWAP gates inserted
+        to perform the qubit routing.
+
+        Args:
+            circuit: circuit to route
+
+        Returns:
+            routed circuit
+
+        Raises:
+            ValueError: routing is impossible
+        """
+        device_graph = nx.Graph(tuple(map(self.get_qubit, edge)) for edge in self.CONNECTIVITY)
+        swap_network = route_circuit(circuit, device_graph, algo_name='greedy')
+        return swap_network.circuit
```

Comment:
The value of the `device` property of the returned circuit is `cirq.UNCONSTRAINED_DEVICE`. Should we update the circuit before returning, so that it is tied to the correct device (i.e. `self`)? 

Updating the device of the circuit results into a call to the `validate_circuit()` method of the new device. So theoretically this might even result into a `ValueError` (if for some reason routing exits normally but does the routing incorrectly). Maybe it is a good idea to catch this exception and re-throw with an updated informative message?

Reply:
We can't, since the routing step may add SWAPs which aren't native to the device, which the user needs to decompose using `decompose_circuit`.